pkg/rotator/rotator.go

```diff
@@ -495,39 +544,101 @@ func (r *ReconcileVWH) Reconcile(request reconcile.Request) (reconcile.Result, e
 		return reconcile.Result{Requeue: true}, err
 	}
 
+	if secret.GetDeletionTimestamp().IsZero() {
+		artifacts, err := buildArtifactsFromSecret(secret)
+		if err != nil {
+			crLog.Error(err, "secret is not well-formed, cannot update webhook configurations")
+			return reconcile.Result{}, nil
+		}
+
+		// Ensure certs on validating webhooks.
+		errVWH := r.ensureVWHCerts(artifacts.CertPEM)
+
+		// Ensure certs on CRD conversion webhooks if there's any.
+		errCWH := r.ensureCRDConvWHCerts(artifacts.CertPEM)
+
+		// Return errors if there's any when trying to inject certs to all webhooks.
+		if errVWH != nil {
+			return reconcile.Result{}, errVWH
+		}
+		if errCWH != nil {
+			return reconcile.Result{}, errCWH
+		}
+
+		// Set CAInjected if it's not early exited.
+		if !r.wasCAInjected.Load() {
```

Comment:
Do we need to test the value here? Should we just store true every time?

Reply:
Ah right, probably it's because I was initially trying to use channel. Will just set true every time.